docs/machine-learning/how-to-guides/inspect-intermediate-data-ml-net.md

```diff
@@ -1,75 +1,180 @@
+        HistoricalPrices = new float[] { 99000f, 98000f, 130000f },
+        CurrentPrice = 180000f
+    }
+};
+```
 
-During the experiment, you may want to observe and validate the data processing results at a given point. This isn't easy since ML.NET operations are lazy, constructing objects that are 'promises' of data.
+## Convert IDataView to IEnumerable
 
-The `GetColumn<T>` extension method lets you inspect the intermediate data. It returns the contents of one data column as an `IEnumerable`.
+One of the simplest ways to inspect the values of an IDataView is to convert it to an [`IEnumerable`](xref:System.Collections.Generic.IEnumerable%601). To convert an [`IDataView`](xref:Microsoft.ML.IDataView) to [`IEnumerable`](xref:System.Collections.Generic.IEnumerable%601) use the [`CreateEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.CreateEnumerable*) method.
 
-The following example shows how to use the `GetColumn<T>` extension method:
+> [!WARNING]
+> Converting the result of [`CreateEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.CreateEnumerable*) to an array or list will load the entire [`IDataView`](xref:Microsoft.ML.IDataView) object into memory which may affect performance.
 
-[Example file](https://github.com/dotnet/machinelearning/tree/master/test/data/adult.tiny.with-schema.txt):
+The code snippet below is useful when you want to access a specific row in the dataset.
+
+```csharp
+// Create an array of HousingData objects from IDataView
+HousingData[] housingDataEnumerable =
+    mlContext.Data.CreateEnumerable<HousingData>(data, reuseRowObject: false).ToArray();
 
-<!-- markdownlint-disable MD010 -->
+// Get first element of IEnumerable
+HousingData firstRow = housingDataEnumerable[0];
 ```
-Label	Workclass	education	marital-status
-0	Private	11th	Never-married
-0	Private	HS-grad	Married-civ-spouse
-1	Local-gov	Assoc-acdm	Married-civ-spouse
-1	Private	Some-college	Married-civ-spouse
 
+When iterating over the entire dataset, loading the contents of the [`IDataView`](xref:Microsoft.ML.IDataView) object into memory isn't the most efficient approach. To optimize performance, set the value of `reuseRowObject` to `true` and leave the result of [`CreateEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.CreateEnumerable*) as an [`IEnumerable`](xref:xref:System.Collections.Generic.IEnumerable%601). This will lazily populate the same object with the data of the current row as it's being evaluated as opposed to creating a new object for each row in the dataset.  
+
+```csharp
+// Create an IEnumerable of HousingData objects from IDataView
+IEnumerable<HousingData> housingDataEnumerable =
+    mlContext.Data.CreateEnumerable<HousingData>(data, reuseRowObject: true);
+
+// Create an IEnumerator to go through housingDataEnumerable
+using (IEnumerator<HousingData> housingDataEnumerator = housingDataEnumerableReuseRowObject.GetEnumerator())
+{
+    // Iterate over housingDataEnumerable
+    while(housingDataEnumerator.MoveNext())
```

Comment:
why not use a foreach-loop instead of creating GetEnumerator() and MoveNext()?

You can replace this whole block with this which is much cleaner:

foreach (HousingData row in housingDataEnumerableReuseRowObject)
     Console.WriteLine(row.Size);

Reply:
#resolved